ami/ml/tests.py

```
@@ -554,6 +554,43 @@ def test_collect_images(self):
+        """
+        collect_images() must hand back SourceImage rows with deployment and
+        deployment.data_source already joined, so the downstream queue_images_to_nats
+        loop doesn't trigger N+1 FK lookups inside image.url() (see issue #1321).
+        """
+        from ami.main.models import S3StorageSource
+
+        data_source = S3StorageSource.objects.create(
+            name="ds-prefetch-test",
+            bucket="prefetch-bucket",
+            access_key="x",
+            secret_key="y",
```

Ruff (0.15.14)

[error] Possible hardcoded password assigned to argument: "secret_key"

(S106)